pkgs/build-support/testers/default.nix

```diff
@@ -97,17 +97,23 @@
   # See doc/builders/testers.chapter.md or
   # https://nixos.org/manual/nixpkgs/unstable/#tester-runNixOSTest
   runNixOSTest =
-    let nixos = import ../../../nixos/lib {
-      inherit lib;
-    };
+    let
+      toGuest = builtins.replaceStrings [ "darwin" ] [ "linux" ];
+      guestPkgs =
+        if (!pkgs.stdenv.hostPlatform.isDarwin)
+          then pkgs
+          else import pkgs.path { system = toGuest pkgs.system; };
+      nixos = import ../../../nixos/lib {
+        inherit lib;
+      };
     in testModule:
         nixos.runTest {
           _file = "pkgs.runNixOSTest implementation";
           imports = [
             (lib.setDefaultModuleLocation "the argument that was passed to pkgs.runNixOSTest" testModule)
           ];
           hostPkgs = pkgs;
-          node.pkgs = pkgs;
+          node.pkgs = guestPkgs;
```

Comment:
Maybe this could be automated in the testing modules instead? That way we'd have one place where this is implemented and documented (ie also for the `nixosTests` attribute, which doesn't use this entrypoint, and I'm not sure whether it should).

Reply:
hm not sure, i was just using it as is, didn't realize that one would consider this bad usage. Similarly, from reading your comment i don't really understand how to approach this.

Reply:
Can be done later.